examples/java/src/test/java/org/apache/beam/examples/WordCountIT.java

```diff
@@ -99,90 +97,62 @@ public void testE2EWordCount() throws Exception {
     }
 
     @Override
-    public boolean matches(Object o) {
-      if (o == null || !(o instanceof PipelineResult)) {
-        fail(String.format("Expected PipelineResult but received %s", o));
-      }
-
+    protected boolean matchesSafely(PipelineResult pResult) {
       if (outputPath == null || outputPath.isEmpty()) {
```

Comment:
Make this a Precondition in the constructor so that we fail fast.
https://google.github.io/guava/releases/19.0/api/docs/com/google/common/base/Preconditions.html